### PR TITLE
CI: travis -> github actions

### PR DESCRIPTION
**Pull request by richardm-stripe**
## Notify
r? @dcr-stripe 
cc @akalinin-stripe @stripe/api-libraries 

## Summary
Migrates the stripe-mock release workflow from Travis to Github Actions. I have set up the appropriate secrets in this repository.

See https://github.com/richardm-stripe/stripe-mock/runs/3292314464?check_suite_focus=true for an example execution on a test repo.

I haven't tested the `docker` or `nfpms` build steps yet. I've changed the docker credentials so that they will point to `stripe/stripe-mock` docker image instead of the `stripemock/stripe-mock`, which has been a longstanding TODO. I mostly expect these steps to fail at first and that I'll have to do another push.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Code checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      -
```
Comment (dcr-stripe) [inline .github/workflows/release.yml:15]: Nit: Merge with line below?

**Review by dcr-stripe — APPROVED** [review]
LGTM

**Comment by akalinin-stripe** [comment]
Looks good to me!

File: .github/workflows/release.yml (hunk 1 above)
Comment (richardm-stripe) [inline .github/workflows/release.yml:15]:
```suggestion
```

**Pull request merged**